ocaml/xcp-rrdd/bin/rrdd/rrdd_server.ml

```diff
@@ -456,6 +457,26 @@ let query_host_ds (ds_name : string) : float =
           query ds_name rrdi
   )
 
+(** Dump all latest data of host dss to file in json format so that any client
+    can read even if it's non-privileged user, such as NRPE. *)
+let dump_host_dss_to_file (file : string) : unit =
+  let json =
+    with_lock mutex (fun () ->
+        match !host_rrd with
+        | None ->
+            `Assoc []
+        | Some rrdi ->
+            `Assoc
+              (Rrd.ds_names rrdi.rrd
+              |> List.map (fun ds_name -> (ds_name, `Float (query ds_name rrdi)))
+              )
+    )
+  in
+  let oc = open_out_gen [Open_wronly; Open_creat; Open_trunc] 0o644 file in
+  finally
+    (fun () -> Yojson.Basic.pretty_to_channel oc json)
+    (fun () -> close_out oc)
```

Comment:
It's fine to not call `close_out_noerr` because our finally deals with exception raised here